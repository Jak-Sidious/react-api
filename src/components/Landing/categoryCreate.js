import React, { Component } from "react";
import Navigation from "../Navbar/navbar";


class Landing extends Component{
  // constructor(props){
  //   super(props);
  //
  //   this.state={
  //
  //   };
  // }

  render() {

    return(
      <div>
      <Navigation/>
      <h1>Tickle</h1>


      </div>
    );
  }
}



export default Landing;
