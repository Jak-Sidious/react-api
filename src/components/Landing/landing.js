// component that renders the landing page
import React, { Component } from 'react';
import Notifications from 'react-notify-toast';
import Navigation from '../Navbar/navbar';

class Landing extends Component {
  render() {
    const { location: { pathname } } = this.props;
    console.log(pathname);
    return (
      <div className="mainBackground">
        <Navigation pathname={pathname} />
        <Notifications />

        <h1 className="Welcome">Welcome to yummy recipes</h1>
        <div className="introText">
          <h2> Inside yummy recipes you are able to do the following </h2>
          <ul>
            <li>Create Food Categories</li>
            <li>Edit Food Categories</li>
            <li>Delete Food Categories</li>
            <li>Create Food Recipes</li>
            <li>Edit Food Recipes</li>
            <li>Delete Food Recipes</li>
          </ul>
        </div>
      </div>
    );
  }
}

export default Landing;
