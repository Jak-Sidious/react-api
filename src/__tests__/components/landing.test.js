import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { MemoryRouter } from 'react-router-dom';

import Landing from '../../components/Landing/landing';


Enzyme.configure({ adapter: new Adapter() });

describe('<Landing />', () => {
  const pathname = 'path';
  const props = {
    location: { pathname }
  };
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <MemoryRouter>
        <Landing {...props} />
      </MemoryRouter>
    );
  });
  it('should render properly', () => {
    const wrapper = shallow(<Landing {...props}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
