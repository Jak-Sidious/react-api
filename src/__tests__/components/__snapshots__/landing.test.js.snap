// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should render properly 1`] = `
<div
  className="mainBackground"
>
  <Navigation
    pathname="path"
  />
  <_class
    options={Object {}}
  />
  <h1
    className="Welcome"
  >
    Welcome to yummy recipes
  </h1>
  <div
    className="introText"
  >
    <h2>
       Inside yummy recipes you are able to do the following 
    </h2>
    <ul>
      <li>
        Create Food Categories
      </li>
      <li>
        Edit Food Categories
      </li>
      <li>
        Delete Food Categories
      </li>
      <li>
        Create Food Recipes
      </li>
      <li>
        Edit Food Recipes
      </li>
      <li>
        Delete Food Recipes
      </li>
    </ul>
  </div>
</div>
`;
